tsdb/shard.go

```diff
@@ -546,26 +546,27 @@ func (s *Shard) validateSeriesAndFields(points []models.Point) ([]models.Point,
 	var skip bool
 	for i, p := range points {
 		skip = false
-		// verify the tags and fields
+
+		// Drop any series w/ a "time" tag, these are illegal
 		tags := p.Tags()
+		skip = tags.Get(timeBytes) != nil
 		if v := tags.Get(timeBytes); v != nil {
-			s.logger.Info(fmt.Sprintf("dropping tag 'time' from '%s'\n", p.PrecisionString("")))
-			tags.Delete(timeBytes)
-			p.SetTags(tags)
+			dropped++
```

Comment:
Given `tags.Get` is `O(n)` you could argue that having a `var skip bool` and then setting `skip` to true inside this if block to reduce the number of calls to `tags.Get` would be worthwhile in some cases.